packages/backend/src/api/controllers/ForcedActionsController.ts

```diff
@@ -0,0 +1,147 @@
+import {
+  renderNewPerpetualForcedTradePage,
+  renderNewSpotForcedWithdrawPage,
+} from '@explorer/frontend'
+import { UserDetails } from '@explorer/shared'
+import { AssetHash, AssetId, EthereumAddress } from '@explorer/types'
+
+import { UserService } from '../../core/UserService'
+import { AssetRepository } from '../../peripherals/database/AssetRepository'
+import { PreprocessedAssetHistoryRepository } from '../../peripherals/database/PreprocessedAssetHistoryRepository'
+import { ControllerResult } from './ControllerResult'
+
+export class ForcedActionsController {
+  constructor(
+    private readonly userService: UserService,
+    private readonly preprocessedAssetHistoryRepository: PreprocessedAssetHistoryRepository<
+      AssetHash | AssetId
+    >,
+    private readonly assetRepository: AssetRepository
+  ) {}
+
+  async getSpotForcedWithdrawalPage(
+    givenUser: Partial<UserDetails>,
+    vaultId: bigint
+  ): Promise<ControllerResult> {
+    const user = await this.userService.getUserDetails(givenUser)
+
+    if (!user) {
+      return { type: 'not found', content: 'User must be logged in' }
+    }
+
+    const assets =
+      await this.preprocessedAssetHistoryRepository.getCurrentByPositionOrVaultId(
+        vaultId
+      )
+
+    const asset = assets[0]
```

Comment:
I think we need @adamiak ‘s opinion on this but from what he told me gammaX doesn’t follow the rule one vault = one asset. Although I may have gotten something wrong.

Reply:
In case of Spot there's only one asset in the vault, so it's ok here.